src/vm/dllimport.cpp

```diff
@@ -6096,7 +6104,42 @@ static void DetermineLibNameVariations(const char* const** libNameVariations, in
         *numberOfVariations = COUNTOF(NameOnly);
     }
 }
-#endif
+#else // FEATURE_PAL
+static void DetermineLibNameVariations(const char* const** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
+{
+    // The purpose of following code is to workaround LoadLibrary limitation: 
+    // LoadLibrary won't append extension if filename itself contains '.'. Thus it will break the following scenario:
+    // [DllImport("A.B")] // The full name for file is "A.B.dll". This is common code pattern for cross-platform PInvoke
+    // The workaround for above scenario is to call LoadLibrary with "A.B" first, if it fails, then call LoadLibrary with "A.B.dll"
+    if (!libName.Find(libName.Begin(), W('.')) || 
+        libName.EndsWith(W(".")) || 
+        libName.EndsWithCaseInsensitive(W(".dll")) || 
+        libName.EndsWithCaseInsensitive(W(".exe")) || 
+        !libNameIsRelativePath)
```

Comment:
`libNameIsRelativePath` should be first because of it is the cheapest condition to evaluate.

Reply:
Also, is the relative path condition really correct here? Relative path is going to be for a path like:

`XXX\YYY\library`

or

`\XXX\YYY\library`.

On Unix, we will end up trying paths like:

`libXXX\YYY\library.so`

or 

`lib\XXX\YYY\library`.

Reply:
Move libNameIsRelativePath  to the first one. 

 'is the relative path condition really correct here?

This is incorrect. since this is an old issue and I will create another issue to track and solve this problem